src/renderer/atlas/AtlasEngine.cpp

```diff
@@ -427,13 +427,13 @@ try
     if (y > hiStart.y)
     {
         const auto isFinalRow = y == hiEnd.y;
-        const auto end = isFinalRow ? std::min(hiEnd.x + 1, x2) : x2;
+        const auto end = isFinalRow ? std::min(hiEnd.x, x2) : x2;
         _fillColorBitmap(row, x1, end, fgColor, bgColor);
 
         // Return early if we couldn't paint the whole region (either this was not the last row, or
         // it was the last row but the highlight ends outside of our x range.)
         // We will resume from here in the next call.
-        if (!isFinalRow || hiEnd.x /*inclusive*/ >= x2 /*exclusive*/)
+        if (!isFinalRow || hiEnd.x > x2)

```

Comment:
BTW I think we may have made a mistake here due to my poor understanding of this code. Probably needs to be mulled over once more...

Reply:
yes, be careful - this is used in TWO places. search highlights and selection rendering. they're both inclusive today. this PR only says it changes one of them.